src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java19/ExpectedGenericRecordDeconstructionPattern.txt

```diff
@@ -0,0 +1,792 @@
+        |       |   |       |       |   |   |--IDENT -> b [17:15]
+        |       |   |       |       |   |   `--IDENT -> v [17:17]
+        |       |   |       |       |   |--ELIST -> ELIST [17:19]
+        |       |   |       |       |   `--RPAREN -> ) [17:19]
+        |       |   |       |       `--LITERAL_NULL -> null [17:24]
+        |       |   |       `--METHOD_CALL -> ( [18:32]
+        |       |   |           |--DOT -> . [18:25]
+        |       |   |           |   |--STRING_LITERAL -> "whatever" [18:15]
+        |       |   |           |   `--IDENT -> equals [18:26]
+        |       |   |           |--ELIST -> ELIST [18:34]
+        |       |   |           |   `--EXPR -> EXPR [18:34]
+        |       |   |           |       `--DOT -> . [18:34]
+        |       |   |           |           |--IDENT -> b [18:33]
+        |       |   |           |           `--IDENT -> v [18:35]
+        |       |   |           `--RPAREN -> ) [18:36]
+        |       |   |--RPAREN -> ) [18:37]
+        |       |   `--LITERAL_RETURN -> return [18:39]
+        |       |       |--EXPR -> EXPR [18:46]
+        |       |       |   `--NUM_INT -> 1 [18:46]
+        |       |       `--SEMI -> ; [18:47]
+        |       |--LITERAL_RETURN -> return [19:8]
+        |       |   |--EXPR -> EXPR [19:15]
+        |       |   |   `--UNARY_MINUS -> - [19:15]
+        |       |   |       `--NUM_INT -> 1 [19:16]
+        |       |   `--SEMI -> ; [19:17]
+        |       `--RCURLY -> } [20:4]
+        |--METHOD_DEF -> METHOD_DEF [22:4]
+        |   |--MODIFIERS -> MODIFIERS [22:4]
+        |   |--TYPE -> TYPE [22:4]
+        |   |   `--LITERAL_INT -> int [22:4]
+        |   |--IDENT -> m3 [22:8]
+        |   |--LPAREN -> ( [22:10]
+        |   |--PARAMETERS -> PARAMETERS [22:11]
+        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [22:11]
+        |   |       |--MODIFIERS -> MODIFIERS [22:11]
+        |   |       |--TYPE -> TYPE [22:11]
+        |   |       |   |--IDENT -> Box [22:11]
+        |   |       |   `--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [22:14]
+        |   |       |       |--GENERIC_START -> < [22:14]
+        |   |       |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [22:15]
+        |   |       |       |   `--IDENT -> String [22:15]
+        |   |       |       `--GENERIC_END -> > [22:21]
+        |   |       `--IDENT -> b [22:23]
+        |   |--RPAREN -> ) [22:24]
+        |   `--SLIST -> { [22:26]
+        |       |--LITERAL_SWITCH -> switch [23:8]
+        |       |   |--LPAREN -> ( [23:15]
+        |       |   |--EXPR -> EXPR [23:16]
+        |       |   |   `--IDENT -> b [23:16]
+        |       |   |--RPAREN -> ) [23:17]
+        |       |   |--LCURLY -> { [23:19]
+        |       |   |--CASE_GROUP -> CASE_GROUP [24:12]
+        |       |   |   |--LITERAL_CASE -> case [24:12]
+        |       |   |   |   |--PATTERN_DEF -> PATTERN_DEF [24:17]
+        |       |   |   |   |   `--RECORD_PATTERN_DEF -> RECORD_PATTERN_DEF [24:17]
```

Comment:
Having these directly nested like this is not good, need to rework the ast a bit

Reply:
Ok, fixed this.